examples/finite_prime_field.rs

```diff
@@ -0,0 +1,254 @@
+    /// is in the correct range by adding the modulus if necessary.
+    fn sub(self, _other: Self) -> Self {
+        Self::new(self.modulus, Self::zero_array())
+    }
+}
+
+impl<const L: usize, const D: usize> SubAssign for FinitePrimeField<L, D> {
+    fn sub_assign(&mut self, _rhs: Self) {
+        todo!()
+    }
+}
+
+impl<const L: usize, const D: usize> PartialEq for FinitePrimeField<L, D> {
+    fn eq(&self, _other: &Self) -> bool {
+        true
+    }
+}
+
+impl<const L: usize, const D: usize> Mul for FinitePrimeField<L, D> {
+    type Output = Self;
+
+    fn mul(self, _other: Self) -> Self {
+        Self::new(self.modulus, Self::zero_array())
+    }
+}
+
+impl<const L: usize, const D: usize> One for FinitePrimeField<L, D> {
+    fn one() -> Self {
+        todo!()
+    }
+}
+
+impl<const L: usize, const D: usize> MulAssign for FinitePrimeField<L, D> {
+    fn mul_assign(&mut self, _rhs: Self) {
+        todo!()
+    }
+}
+
+impl<const L: usize, const D: usize> Inv for FinitePrimeField<L, D> {
+    type Output = Self;
+
+    fn inv(self) -> Self {
+        Self::new(self.modulus, Self::zero_array())
+    }
+}
+
+impl<const L: usize, const D: usize> Euclid for FinitePrimeField<L, D> {
+    fn div_euclid(&self, _v: &Self) -> Self {
+        todo!()
+    }
+
+    fn rem_euclid(&self, _v: &Self) -> Self {
+        todo!()
+    }
+}
+
+impl<const L: usize, const D: usize> Rem for FinitePrimeField<L, D> {
+    type Output = Self;
+
+    fn rem(self, _rhs: Self) -> Self::Output {
+        todo!()
+    }
+}
+
+impl<const L: usize, const D: usize> Div for FinitePrimeField<L, D> {
+    type Output = Self;
+
+    fn div(self, _other: Self) -> Self {
+        Self::new(self.modulus, Self::zero_array())
+    }
+}
+
+impl<const L: usize, const D: usize> DivAssign<Self> for FinitePrimeField<L, D> {
+    fn div_assign(&mut self, _rhs: Self) {
+        todo!()
+    }
+}
+
+/// Marker trait for commutative addition: a + b = b + a
+impl<const L: usize, const D: usize> CommutativeAddition for FinitePrimeField<L, D> {}
+
+/// Marker trait for commutative multiplication: a * b = b * a
+impl<const L: usize, const D: usize> CommutativeMultiplication for FinitePrimeField<L, D> {}
+
+/// Marker trait for associative addition: (a + b) + c = a + (b + c)
+impl<const L: usize, const D: usize> AssociativeAddition for FinitePrimeField<L, D> {}
+
+/// Marker trait for associative multiplication: (a * b) * c = a * (b * c)
+impl<const L: usize, const D: usize> AssociativeMultiplication for FinitePrimeField<L, D> {}
+
+/// Marker trait for distributive multiplication over addition: a * (b + c) = (a * b) + (a * c)
+impl<const L: usize, const D: usize> Distributive for FinitePrimeField<L, D> {}
+
+impl<const L: usize, const D: usize> FiniteField for FinitePrimeField<L, D> {
+    fn characteristic() -> u64 {
+        todo!()
+    }
+
+    fn order() -> u64 {
+        todo!()
+    }
+
+    fn generator() -> Self {
```

Comment:
Presumably an obvious choice for a generator since there can be more than one?

Reply:
Yeah, I guess that's true. I think this function can go. Ideally it'd return all generators, but I don't actually have a use case thought up for that quite yet. I also am trying to just provide an idiomatic template for implementation using the standard traits more that getting proscriptive about which functionalities must be implemented.

Reply:
Removed function.